### PR TITLE
JSX -> Babel cleanup

### PR DESCRIPTION
**Pull request by zpao**
- Removed the old transforms directory from the test list (apparently caused issues on Windows)
- Used Babel for docs code generation (as well as moved some files away so they wouldn't get transformed).

### CODE REVIEW

File: docs/Rakefile [hunk 1]
```diff
@@ -4,12 +4,12 @@ require('yaml')
 
 desc "generate js from jsx"
 task :js do
-  system "../bin/jsx --harmony _js js"
+  system "../node_modules/.bin/babel _js --out-dir=js"
 end
 
 desc "watch js"
 task :watch do
```
Comment (spicyj) [inline docs/Rakefile]: TIL this exists

File: package.json [hunk 2]
```diff
@@ -29,7 +29,8 @@
     "jstransform": "^11.0.0"
   },
   "devDependencies": {
-    "babel-core": "^5.2.7",
+    "babel": "^5.3.1",
+    "babel-core": "^5.3.1",
```
Comment (spicyj) [inline package.json]: So these don't get out of sync, want to just replace our uses of `require('babel-core')` with `require('babel')`? Their node API is identical.

Comment (zpao) [inline package.json]: Yea, I thought about it but couldn't find anything that actually said it was the same (without looking at src)

File: docs/Rakefile (hunk 1 above)
Comment (zpao) [inline docs/Rakefile]: It was really useful in the early days, not so much anymore.

File: package.json (hunk 2 above)
Comment (spicyj) [inline package.json]: http://babeljs.io/docs/usage/api/#require-babel-core-

Comment (zpao) [inline package.json]: https://www.npmjs.com/package/babel is devoid of anything useful :/

**Pull request merged**